packages/airnode-node/src/evm/gas-prices.ts

```diff
@@ -1,20 +1,49 @@
 import { ethers } from 'ethers';
 import { go } from '../utils/promise-utils';
 import * as logger from '../logger';
-import { DEFAULT_RETRY_TIMEOUT_MS } from '../constants';
-import { LogsData } from '../types';
+import { BASE_FEE_MULTIPLIER, DEFAULT_RETRY_TIMEOUT_MS, PRIORITY_FEE, WEI_PER_GWEI } from '../constants';
+import { GasTarget, LogsData, PendingLog } from '../types';
 
 interface FetchOptions {
   readonly provider: ethers.providers.JsonRpcProvider;
 }
 
-export async function getGasPrice(options: FetchOptions): Promise<LogsData<ethers.BigNumber | null>> {
+export const getGasPrice = async (options: FetchOptions): Promise<LogsData<GasTarget | null>> => {
   const { provider } = options;
+  const [logs, getBlockGas] = await (async (): Promise<[PendingLog[], GasTarget | undefined]> => {
+    const operation = () => provider.getBlock('latest');
+    const [err, blockHeader] = await go(operation, { retries: 1, timeoutMs: DEFAULT_RETRY_TIMEOUT_MS });
+    if (err || !blockHeader?.baseFeePerGas) {
+      const log = logger.pend('INFO', 'Failed to get EIP-1559 gas pricing from provider - trying fallback', err);
+
+      return [[log], undefined];
```

Comment:
This returns `undefined` but elsewhere returns `null`?

Reply:
It won't have a practical effect because the `undefined` value triggers an attempt to retrieve the fallback (which results in a valid value or null), but I'll swap it out for consistency.